src/ingestion/models/ingestionManager.ts

```diff
@@ -160,78 +172,93 @@ export class IngestionManager {
       isCleaned: false,
       shouldReturnTasks: false,
     };
     const jobs = await this.jobManagerWrapper.getJobs<Record<string, unknown>, ITaskParameters>(findJobParameters);
-    if (requestType === RequestType.NEW) {
-      jobs.forEach((job) => {
-        if (job.status == OperationStatus.IN_PROGRESS || job.status == OperationStatus.PENDING) {
-          const message = `Layer id: ${productId} product type: ${productType}, conflicting job ${job.type} is already running for that layer`;
-          this.logger.error({
-            productId: productId,
-            productType: productType,
-            msg: message,
-            logCtx: logCtx,
-          });
-          throw new ConflictError(message);
-        }
-      });
-    } else {
-      const ingestionJobTypes = this.config.get<string[]>('forbiddenTypesForParallelIngestion');
-      jobs.forEach((job) => {
-        if ((job.status == OperationStatus.IN_PROGRESS || job.status == OperationStatus.PENDING) && ingestionJobTypes.includes(job.type)) {
-          const message = `Layer id: ${productId} product type: ${productType}, conflicting job ${job.type} is already running for that layer`;
-          this.logger.error({
-            productId: productId,
-            productType: productType,
-            msg: message,
-            logCtx: logCtx,
-          });
-          throw new ConflictError(message);
-        }
-      });
-    }
+    return jobs;
   }
 
-  private async isInMapProxy(productId: string, productType: ProductType, requestType: RequestType): Promise<void> {
-    const logCtx: LogContext = { ...this.logContext, function: this.isInMapProxy.name };
-    const layerName = getMapServingLayerName(productId, productType);
-    const existsInMapServer = await this.mapProxyClient.exists(layerName);
-    if (existsInMapServer && requestType === RequestType.NEW) {
-      const message = `Failed to create new ingestion job for layer: '${layerName} ', already exists on MapProxy`;
+  private async validateLayerDoesntExistInMapProxy(productId: string, productType: ProductType): Promise<void> {
+    const logCtx: LogContext = { ...this.logContext, function: this.validateLayerDoesntExistInMapProxy.name };
+    const exists = await this.getlLayerExistanceInMapProxy(productId, productType);
+    if (exists) {
+      const message = `Failed to create new ingestion job for layer: ${productId}-${productType}, already exists on MapProxy`;
       this.logger.error({
         productId: productId,
         productType: productType,
-        mapProxyLayerName: layerName,
         msg: message,
         logCtx: logCtx,
       });
       throw new ConflictError(message);
-    } else if (!existsInMapServer && requestType === RequestType.UPDATE) {
-      const message = `Failed to create update job for layer: '${layerName} ', layer doesnt exist on MapProxy`;
+    }
+  }
+
+  private async validateLayerExistsInMapProxy(productId: string, productType: ProductType): Promise<void> {
```

Comment:
check if you can pass here just the maproxyLayerId instead productId and productType

Reply:
why? the function creates the layer name inside. creating one line functions is not a good practice

Reply:
becuase you are passing 2 params that are only used once inside to get one parameter - it's just passing more parameters than you should

Reply:
you asked for this implementation a week ago. but ok, removed it.
you told me this makes the code more readable and neat and i agreed

Reply:
done with passing one param